File: Native/Test/Node/mr-array-transform.js
/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

FC = require("Fabric").createClient();

cv = FC.MR.createConstValue("Size", 10);

ago = FC.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Scalar output, Size index) { output = Scalar(index); }", "foo");
ag = FC.MR.createArrayGenerator(cv, ago);

to = FC.KLC.createArrayTransformOperator("bar.kl", "operator bar(io Scalar value) { value *= 3.14; }", "bar");
t = FC.MR.createArrayTransform(ag, to);

var count = t.getCount();
console.log("t.getCount() = " + count);
for (var i=0; i<count; ++i)
  console.log("t.produce("+i+") = "+t.produce(i));

FC.close();

File: Native/Test/Python/mr-array-transform.py
#
#  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
#

import fabric

client = fabric.createClient()

cv = client.MR.createConstValue("Size", 10)

ago = client.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Scalar output, Size index) { output = Scalar(index); }", "foo")
ag = client.MR.createArrayGenerator(cv, ago)

to = client.KLC.createArrayTransformOperator("bar.kl", "operator bar(io Scalar value) { value *= 3.14; }", "bar")
t = client.MR.createArrayTransform(ag, to)

count = t.getCount()
print("t.getCount() = " + str(count))
for i in range(0, count):
  print("t.produce("+str(i)+") = "+str(round(t.produce(i),3)))

client.close()
